miriway

```diff
@@ -1,6 +1,24 @@
 #! /bin/sh
 set -e
 
+### Gtk Theming Tweaks
+gtk30_config="${XDG_CONFIG_HOME:-$HOME/.config}/gtk-3.0/settings.ini"
+if [ -f "$gtk30_config" ]; 
+then
+  cursor_theme="$(grep 'gtk-cursor-theme-name' "$gtk30_config" | sed 's/.*\s*=\s*//')"
+  font_name="$(grep 'gtk-font-name' "$gtk30_config" | sed 's/.*\s*=\s*//')"
+  gtk_theme="$(grep 'gtk-theme-name' "$gtk30_config" | sed 's/.*\s*=\s*//')"
+  icon_theme="$(grep 'gtk-icon-theme-name' "$gtk30_config" | sed 's/.*\s*=\s*//')"
+  
+  gsettings set org.gnome.desktop.interface gtk-theme "$gtk_theme"
+  gsettings set org.gnome.desktop.interface icon-theme "$icon_theme"
+  gsettings set org.gnome.desktop.interface cursor-theme "$cursor_theme"
+  gsettings set org.gnome.desktop.interface font-name "$font_name"
+fi
+
+### Window Buttons Fix
+gsettings set org.gnome.desktop.wm.preferences button-layout ":minimize,maximize,close"
```

Comment:
We want to start miriway on systems without `gsettings`, so please make this all conditional on `gsettings` being available:

```bash
if ! command -v gsettings > /dev/null
```
etc.

Reply:
Regarding your concerns , gtk apps cant be modified without gsettings in wayland, one can use xsettingsd but its confined to xorg, remaing for the gtk3-0 file, i or any program need something like a file to understand the user choice, gtk3-0 is one of the most commonly file found in most of the distributions supporting gtk 3 theming including lxqt and Kde, if ypur system doesnt have that than we will need some other file to understand and implement user theming choice, you can also write a GUI or CLI App specific to miriway seeking inputs for user theming specs like nwg-look  if you dont prefer the above method, the only best and sensible alternative to gtk3-0 config file on Gnome desktop is to fetch information from gsettings get function and then do the  theming but as per my knowledge currently only gnome exclusively relies on gsettings its a newer method of theming in wayland  currently only used by gnome as such if miriway is used alongside kde/lxqt/xfce/mate where a user doesnt have configurations with gsettings the tweaks will fail , i will work on making gsettings conditional

Reply:
@AlanGriffiths  i hope you read my comment above, now regarding your final concern that wont it mess up gnome if these keys are there, then let me tell you the config file from which my tweaks fetch user choice configures data is also the same one in which the user gtk 3 theming choice is stored by other DEs like lxqt/mate/xfce/kde etc. as such its actually forcing miriway to actually uniformly theme same as the another DE its installed alongwith, but in the case of modern gnome (maybe even your one) where gsettings are explicitly used the gtk3 config maynot exist as such miriway should theme itself if gsettings is installed but this will ony happen in case if miriway is installed along with Gnome or other DEs , in case of an individual installation , the developer or the user must install gsettings and then manually enter the gsettings command as mentioned in the tweak or simply create a gtk-3.0 config file with his/her choice under ~/.config/gtk3-0, and miriway will simply implement his/her choice at every session, i have done the conditional part

Reply:
Let me paraphrase to check I've understood your response:

> 1.  None of my systems has .../gtk-3.0/settings.ini, so when does this do anything?

This file exists with lxqt and Kde; and, otherwise, the user may create it

> 2.    It is changing the "org.gnome.desktop.interface" schema. I would expect that to be specific to GNOME desktop and not be relevant to Miriway, so is this right? Or is there a better way?

You don't know, and, if I don't like it, you expect me to investigate

> 3.    If GNOME desktop already has settings for these keys, won't this mess up GNOME?

This will only matter if GNOME is installed

Have I missed something?

Reply:
@AlanGriffiths  ***You have again Misunderstood, Read Carefully with a Peaceful and Calm mind***
 
1) The gtk3-0 config will exist in any DE that support gtk3-0 theming including Mate,Xfce and Even Sway WM through nwg-look, to theme gtk3-0 in any other desktop environment other than Gnome (41 +)  the config file must exist otherwise you cant theme gtk3-0 apps.

2) Without Gnome components like gsettings, the config file etc. you cant theme gtk apps, you need gnome tools like gsettings to theme gtk apps in wayland, in xorg you can also use xsettingsd, you cant make miriway theme gtk apps without gsettings in wayland.

3) This keys wont mess up Gnome even if its preinstalled because as i said and modified it to work only if gtk3-0 config exist and looking at your response that the file doesnt exist, it wont even change any thing and even if the file exists it will just re-implement the gsettings keys exclusively for miriway as the gsettings key would be similar to that of the config file

Reply:
Let me give an example as I've now found a system which has this `settings.txt` file:

```bash
$ cat /home/alan/.config/gtk-3.0/settings.ini
[Settings]
gtk-enable-primary-paste=true
```

Running your code does change the GNOME desktop settings: E.g.

```bash
~$ gsettings get org.gnome.desktop.interface cursor-theme
'Adwaita'
```

is changed to:

```bash
$ gsettings get org.gnome.desktop.interface cursor-theme
''
```

I think this is the wrong thing to be doing.

Reply:
I wasnt frustrating was just strongly addressing the concerns, sorry if you felt it that way, anyway i am closing this.